### PR TITLE
feat(ci): live deployment for main

### PR DESCRIPTION
**Pull request by tthvo**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] Signed the last commit: `git commit --amend --signoff`
_______________________________________________

Fixes: #1027 

## Description of the change:

Added workflow to build preview assets for PR and main push.
- Assets are build with Mirage JS injected.
- Published with surge -> **Need to add secrets.SURGE_TOKEN**

## Motivation for the change:

See #1027 

## Example:

**On push to main**:

- Repository: https://github.com/tthvo/cryostat-web/
- Preview URL: [https://tthvo-main.surge.sh](https://tthvo-main.surge.sh)
- Action Run: https://github.com/tthvo/cryostat-web/actions/runs/5160629039/jobs/9296850666

### CODE REVIEW

File: .github/workflows/ci-preview.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-preview:
+    if: ${{ github.repository_owner == 'cryostatio' }}
+    env:
+      DEPLOY_DOMAIN: ${{ github.repository_owner }}-main.surge.sh
```
Comment (andrewazores) [inline .github/workflows/ci-preview.yml:16]: Perhaps this should be `cryostatio-cryostat-web-main` or something like that, ie `owner-repo-branch`. I can imagine more cases where we deploy things to Surge, like API docs for various other components.

File: .github/workflows/ci-preview.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-preview:
+    if: ${{ github.repository_owner == 'cryostatio' }}
+    env:
+      DEPLOY_DOMAIN: ${{ github.repository_owner }}-cryostat-web-main.surge.sh
```
Comment (andrewazores) [inline .github/workflows/ci-preview.yml:16]: `{{ github.event.repository.name }}` should work too instead of hardcoding the repo name, I think

Comment (andrewazores) [reply .github/workflows/ci-preview.yml:16]: https://stackoverflow.com/questions/62803531/repository-name-as-a-github-action-environment-variable

File: .github/workflows/ci-preview.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: Generate Preview (CI)
+
+concurrency:
+  group: ci-${{ github.event.number }}
```
Comment (andrewazores) [inline .github/workflows/ci-preview.yml:4]: I think just `group: ci` is fine here, the event number concurrency group is useful when multiple jobs could stem from logically related events like pushes to a PR. But here this should just be happening once on each merge to `main`.

**Comment by tthvo** [comment]
I am using to include any source files, assets, webpack and dependencies changes but exclude tests. Have to use `!` here with `paths` because `paths-ignore` can't be used at the same time with `paths`: 

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-and-excluding-paths

```yaml
    paths:
      - 'src/app/**'
      - 'webpack.*.js'
      - 'yarn.lock'
      - 'tsconfig.json'
      - '!src/test/**'
```

**Review by andrewazores — APPROVED** [review]

**Pull request merged**